packages/api/lib/writeRecords/write-granules.js

```diff
@@ -384,102 +280,182 @@ const _generateFilesFromGranule = async ({
+ * @param {string} [params.endingDateTime] - CMR Echo10: Temporal.RangeDateTime.EndingDateTime
+ * @param {string} [params.productionDateTime] - CMR Echo10: DataGranule.ProductionDateTime
+ * @param {string} [params.lastUpdateDateTime] - CMR Echo10: LastUpdate || InsertTime
+ * @param {string} [params.processingStartDateTime] - execution startDate
+ * @param {string} [params.processingEndDateTime] - execution StopDate
+ * @param {Object} [params.queryFields] - query fields
+ * @param {Object} [params.granuleModel] - only for testing.
+ * @param {Object} [params.granulePgModel] - only for testing.
+ * @param {number} collectionCumulusId - PostGreSQL cumulus_id for this granule's collection
+ * @param {Knex} knex - knex Client
+ * @returns {Promise}
+ */
+const writeGranuleFromApi = async (
+  {
+    granuleId,
+    collectionId,
+    status,
+    execution,
+    cmrLink,
+    published,
+    pdrName,
+    provider,
+    error = {},
+    createdAt = new Date().valueOf(),
+    updatedAt = new Date().valueOf(),
+    duration,
+    productVolume,
+    timeToPreprocess,
+    timeToArchive,
+    files = [],
+    beginningDateTime,
+    endingDateTime,
+    productionDateTime,
+    lastUpdateDateTime,
+    processingStartDateTime,
+    processingEndDateTime,
+    queryFields,
+    granuleModel = new Granule(),
+    granulePgModel = new GranulePgModel(),
+  },
+  collectionCumulusId,
+  knex
+) => {
+  try {
+    // Build a objects with correct shape for the granuleModel.generateGranuleRecord.
+    const granule = { granuleId, cmrLink, published, files };
+    const processingTimeInfo = {
+      processingStartDateTime,
+      processingEndDateTime,
+    };
+    const cmrTemporalInfo = {
+      beginningDateTime,
+      endingDateTime,
+      productionDateTime,
+      lastUpdateDateTime,
+    };
 
-    return granuleModel.storeGranuleFromCumulusMessage({
+    let executionCumulusId;
+    if (execution) {
+      executionCumulusId = await getExecutionCumulusId(execution, knex);
+      if (executionCumulusId === undefined) {
+        throw new Error(`Could not find execution in PostgreSQL database with url ${execution}`);
+      }
+    }
+
+    const dynamoGranuleRecord = await granuleModel.generateGranuleRecord({
       granule,
-      executionUrl,
+      executionUrl: execution,
       collectionId,
       provider,
-      workflowStartTime,
+      timeToArchive,
+      timeToPreprocess,
+      productVolume,
+      duration,
+      status,
+      workflowStartTime: createdAt,
+      files,
       error,
       pdrName,
-      workflowStatus,
-      processingTimeInfo,
       queryFields,
+      processingTimeInfo,
       updatedAt,
+      cmrTemporalInfo,
     });
-  });
 
-  await _writeGranuleFiles({
-    files,
-    granuleCumulusId,
-    granule,
-    workflowError: error,
-    workflowStatus,
-    knex,
-    granuleModel,
-  });
+    const postgresGranuleRecord = await translateApiGranuleToPostgresGranule(
```

Comment:
So I just realized that this translate function is going to look up the `collection_cumulus_id` anyway. So at minimum that's two lookups for it: one in the endpoint logic and one in this function. And furthermore, the `collectionCumulusId` that we're passing into `_writeGranule` is completely unnecessary - it's not used in any write logic, just a log statement. The same information is already available in `postgresGranuleRecord.collection_cumulus_id`

But this may all be irrelevant since I think your endpoint should be checking DynamoDB and not PostgreSQL for an existing record

Reply:
Yeah, I will pull out the collectionCumulusId from writeGranuleFromApi and downwards when I switch to dynamoDB.